guake/guake_app.py

```diff
@@ -1042,8 +1043,24 @@ def recompute_tabs_titles(self):
             page_num = self.get_notebook().page_num(terminal.get_parent())
             self.get_notebook().rename_page(page_num, self.compute_tab_title(terminal), False)
 
+    def read_current_guake_yaml(self, vte):
+
+        cwd = Path(vte.get_current_directory())
+        guake_yaml = cwd.joinpath(".guake.yml")
+        if guake_yaml.is_file():
+            with guake_yaml.open() as fd:
+                return yaml.safe_load(fd)
+
+        return None
+
     def compute_tab_title(self, vte):
         """Compute the tab title"""
+
+        guake_yml = self.read_current_guake_yaml(vte)
+        if guake_yml is not None:
```

Comment:
Maybe we can move these checks into the yaml read function? So if there's more added to the yaml file in the future, there's a bit less boilerplate checking that needs to be rewritten each time.

Reply:
I moved the `is not None` internally to the `read_current_guake_yaml` method. That method is returning a dictionary with the file's content, not just a string with the title, so a minimal check is still required here. I did this thinking that the `.guake.yml` file might contain more than a title.